presto-hive/src/test/java/com/facebook/presto/hive/TestBackgroundHiveSplitLoader.java

```diff
@@ -180,6 +181,105 @@ public void testEmptyFileWithNoBlocks()
         assertEquals(splits.get(0).getLength(), 0);
     }
 
+    @Test
+    public void testNoHangIfPartitionIsOffline()
+            throws Exception
+    {
+        BackgroundHiveSplitLoader backgroundHiveSplitLoader = backgroundHiveSplitLoaderNoMetadata();
+        HiveSplitSource hiveSplitSource = hiveSplitSource(backgroundHiveSplitLoader, TupleDomain.none());
+        backgroundHiveSplitLoader.start(hiveSplitSource);
+
+        try {
+            drain(hiveSplitSource);
+        }
+        catch (Exception e) {
+            assertEquals(e.getMessage(), "java.lang.Exception: OFFLINE");
+        }
+
+        boolean failed = false;
+        try {
+            hiveSplitSource.isFinished();
+        }
+        catch (Exception e) {
+            failed = true;
+        }
+        assertEquals(failed, true);
+    }
+
+    private static HivePartitionMetadata verifyOnline(HivePartitionMetadata hivePartitionMetadata)
+            throws Exception
+    {
+        if (hivePartitionMetadata.getHivePartition().getTableName().getSchemaName().equalsIgnoreCase("testSchema2")) {
+            throw new Exception("OFFLINE");
+        }
+
+        return hivePartitionMetadata;
+    }
+
+    private static Iterable<HivePartitionMetadata> getIterableHivePartitionMetadata()
```

Comment:
name of this method is very generic, while this not a generic `Iterable<HivePartitionMetadata>`, see usage of method `verifyOnline`. It looks like this method could be used only in particular tests so I would suggest you to reflect that in this method name. Maybe `createPartitionMetadataWithOffilnePartitions`?

The same comment applies to the method below.